examples/mongocxx/mongodb.com/documentation_examples.cpp

```diff
@@ -1180,6 +1193,85 @@ void delete_examples(mongocxx::database db) {
     }
 }
 
+static void snapshot_examples(mongocxx::client& client) {
+    // Write concern "majority" needed to avoid the following error below:
+    //
+    // Unable to read from a snapshot due to pending collection catalog changes;
+    // please retry the operation.
+    //
+    // See also: https://jira.mongodb.org/browse/SERVER-41532
+
+    // Start Example 59
+    using namespace mongocxx::v_noabi;
+    using bsoncxx::builder::basic::kvp;
+    using bsoncxx::builder::basic::make_document;
+
+    options::insert write_options;
+    write_concern wc;
+    wc.majority(std::chrono::seconds(10));
+    write_options.write_concern(wc);
+
+    auto db = client["pets"];
+
+    // seed 'pets' database with dogs and cats
+    db.drop();
+    db["cats"].insert_one(make_document(kvp("adoptable", true)), write_options);
+    db["dogs"].insert_one(make_document(kvp("adoptable", true)), write_options);
+    db["dogs"].insert_one(make_document(kvp("adoptable", false)), write_options);
+
+    int64_t adoptable_pets_count = 0;
+
+    auto opts = mongocxx::options::client_session{};
+    opts.snapshot(true);
+    auto session = client.start_session(opts);
+
+    {
+        pipeline p;
+
+        p.match(make_document(kvp("adoptable", true))).count("adoptableCatsCount");
+        auto cursor = db["cats"].aggregate(session, p);
+
+        for (auto doc : cursor) {
+            adoptable_pets_count += doc.find("adoptableCatsCount")->get_int32();
+        }
+    }
+
+    {
+        pipeline p;
+
+        p.match(make_document(kvp("adoptable", true))).count("adoptableDogsCount");
+        auto cursor = db["dogs"].aggregate(session, p);
+
+        for (auto doc : cursor) {
+            adoptable_pets_count += doc.find("adoptableDogsCount")->get_int32();
+        }
+    }
+
+    if (adoptable_pets_count != 2) {
+        throw std::logic_error("wrong number of adoptable pets in example 57, expecting 2 got: " +
+                               std::to_string(adoptable_pets_count));
+    }
+    // End Example 59
+}
```

Comment:
I missed in the C review. `// Start Snapshot Query Example 2` also needs to be added.